client/src/components/MultiJurisdictionAudit/Progress/ProgressMap.tsx

```diff
@@ -0,0 +1,353 @@
+import { IAuditSettings } from '../useAuditSettings'
+
+interface IProps {
+  stateName: string | null
+  jurisdictions: IJurisdiction[]
+  isRoundStarted: boolean
+  auditType: IAuditSettings['auditType']
+}
+
+const MapWrapper = styled.div`
+  position: relative;
+`
+
+const SVGMap = styled.svg`
+  margin-bottom: 30px;
+  .outline {
+    fill: none;
+    stroke: ${Colors.BLACK};
+    stroke-width: 0.5px;
+  }
+  .mesh {
+    fill: none;
+    stroke-linejoin: round;
+  }
+  path {
+    fill: none;
+    stroke: none;
+    stroke-width: 0.5px;
+  }
+  #single-state {
+    stroke: ${Colors.BLACK};
+  }
+  .county {
+    fill: ${Colors.WHITE};
+    stroke: ${Colors.BLACK};
+  }
+  .county:hover {
+    stroke-width: 2px;
+  }
+  .danger {
+    fill: ${Colors.RED3};
+  }
+  .success {
+    fill: ${Colors.GREEN4};
+  }
+  .progress {
+    fill: ${Colors.ORANGE4};
+  }
+  .gray {
+    fill: ${Colors.GRAY4};
+  }
+  .default {
+    fill: ${Colors.WHITE};
+  }
+`
+
+const Tooltip = styled.div`
+  display: none;
+  position: absolute;
+  opacity: 0.9;
+  border-radius: 3px;
+  background: ${Colors.BLACK};
+  padding: 5px;
+  color: ${Colors.WHITE};
+`
+
+const MapLabels = styled.div`
+  position: absolute;
+  top: 0;
+  z-index: 10;
+`
+
+const MapLabelsRow = styled.div`
+  display: flex;
+  align-items: center;
+  margin-bottom: 5px;
+`
+
+const MapLabelsBoxes = styled.div`
+  display: inline-block;
+  margin-right: 10px;
+  width: 20px;
+  height: 20px;
+  &.danger {
+    background-color: ${Colors.RED3};
+  }
+  &.success {
+    background-color: ${Colors.GREEN4};
+  }
+  &.progress {
+    background-color: ${Colors.ORANGE4};
+  }
+  &.gray {
+    background-color: ${Colors.GRAY4};
+  }
+  &.default {
+    border: 1px solid ${Colors.BLACK};
+  }
+`
+
+const MapSpinner = styled(Spinner)`
+  position: absolute;
+  top: 50%;
+  right: 0;
+  left: 0;
+  transform: translateY(-50%);
+  margin: 0 auto;
+`
+
+const Map = ({
+  stateName,
+  jurisdictions,
+  isRoundStarted,
+  auditType,
+}: IProps) => {
+  const width = 960
+  const height = 500
+  const d3Container = useRef(null)
+  const tooltipContainer = useRef(null)
+
+  const [jsonData, setJsonData] = useState<Topology | undefined>(undefined)
+  const [matchedJurisdictions, setMatchedJurisdictions] = useState<boolean>(
+    false
+  )
+
+  const getStateName = (abbr: string) => {
+    const filteredState = labelValueStates.find(
+      ({ value, label }) => value === abbr && label
+    )
+    return filteredState && filteredState.label
+  }
+
+  const getJurisdictionStatusClass = useCallback(
+    (countyName: string) => {
+      const filteredJurisdiction = jurisdictions.find(jurisdiction => {
+        const jurisdictionNameLower = jurisdiction.name.toLowerCase()
+        const countyNameLower = countyName.toLowerCase()
+        return (
+          jurisdictionNameLower === countyNameLower ||
+          jurisdictionNameLower.replace(' county', '') === countyNameLower
+        )
+      })
```

Comment:
I'm noticing that the logic that matches jurisdiction names to county names is duplicated here and below. Maybe a way to simplify would be to compute a mapping from county name to jurisdiction in the beginning, and then use that throughout the component.

```ts
const countyNameToJurisdiction = Object.fromEntries(
  countyNames.map(countyName =>
    [countyName, jurisdictions.find(jurisdiction =>
      jurisdiction.name.toLowerCase().replace(/\s+county(\s+|$)/i, '') === countyName.toLowerCase()
    )]
  )
)

// To get a jurisdiction by county name
countyNameToJurisdiction[countyName]

// To check how many jurisdictions matched a county
Object.values(countyNameToJurisdiction).filter(jurisdiction => jurisdiction !== null).length